modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/pipelining/HttpPipeliningHandler.java

```diff
@@ -120,4 +121,13 @@ public void write(final ChannelHandlerContext ctx, final Object msg, final Chann
         }
     }
 
+    @Override
+    public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+        ClosedChannelException closedChannelException = new ClosedChannelException();
```

Comment:
Okay. I will make a change based on that. And I agree that synchronization is not necessary (as it should be unnecessary in the write method). I just was confused by what you meant by guard.